apps/polyflix/src/modules/attachments/components/AttachmentSelectorModal.component.tsx

```diff
@@ -0,0 +1,210 @@
+  attachments,
+  videoId,
+  isOpen,
+  onClose,
+  sx: sxProps,
+}: Props) => {
+  const { user } = useAuth()
+  const { t } = useTranslation('attachments')
+
+  const [page] = useState(1)
+
+  const [filters, setFilters] = useState<AttachmentParams>({
+    page,
+    pageSize: 10,
+    userId: user!.id,
+  })
+
+  const { data, isLoading, refetch } = useGetUserAttachmentsQuery(filters)
+
+  const { fields, append, remove } = attachments
+
+  const handleToggle = (attachment: Attachment) => () => {
+    const currentIndex = fields.findIndex((e) => e.id === attachment.id)
+    if (currentIndex === -1) {
+      append(attachment)
+    } else {
+      remove(currentIndex)
+    }
+  }
+
+  useEffect(() => {
+    /* Since the attachments are not invalidated after a video update, we need to refetch them here */
+    if (data) refetch()
+  }, [])
+
+  useEffect(() => {
+    /* On update mode, append the previously selected attachments in order to set the checkboxes ticked */
+    if (videoId && data) {
+      for (const f of fields) {
+        if (
+          f.videos.includes(videoId) &&
+          !fields.find(({ id }) => id === f.id)
+        ) {
+          append(f)
+        }
+      }
+    }
+  }, [videoId, data])
+
+  const isAttachmentSelected = (attachment: Attachment) =>
+    fields.some((e) => e.id === attachment.id)
+
+  /* If the user has no attachment, he is redirected to the attachment creation form */
+  if (data && data.totalCount === 0)
+    return <Redirect push to="/users/profile/attachments/create" />
+
+  return (
+    <Modal
+      sx={{
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        padding: 2,
+        ...sxProps,
+      }}
+      open={isOpen}
+      onClose={() => onClose()}
+      aria-labelledby="element modal"
+      closeAfterTransition
+      BackdropProps={{
+        timeout: 500,
+      }}
+    >
+      <Fade in={isOpen}>
+        <Paper
+          sx={{
+            width: {
+              lg: '40%',
+              md: '50%',
+              sm: '70%',
+              xs: '90%',
+            },
+            bgcolor: 'background.default',
+            borderRadius: 2,
+            p: {
+              sm: 2,
+              xs: 1,
+            },
+          }}
+          variant="outlined"
+        >
+          <Typography variant="h4" sx={{ mb: '2%' }}>
+            {t('forms.selector.title')}
+          </Typography>
+          <Scrollbar
+            sx={{
+              maxHeight: (theme) => `calc(100vh - ${theme.spacing(30)})`,
+              minHeight: '300px',
+            }}
+          >
+            <Box sx={{ mt: 2 }}>
+              {{ data }
+                ? data?.items.map((item) => (
+                    <ListItem
+                      key={item.id}
+                      secondaryAction={
+                        <Checkbox
+                          edge="end"
+                          onChange={handleToggle(item)}
```

Comment:
I think it is better like that if you use the input of the function handleToggle

```suggestion
                          onChange={() => handleToggle(item)}
```